### DISCUSSION TITLE
Setting multiple extensions for figures?

### DISCUSSION THREAD

**Discussion by ABSOD** [quarto-dev/quarto-cli#9276] [Q&A]
### Description

I am using Quarto with R. to produce a PDF document  I need to insert a figure from an external file and set multiple extensions (the width of the figure and an alt-text string)
The example for figures shows this syntax:

`![Elephant](elephant.png){width=4in}`

Is it possible to include more than one extension, e.g.:

`![Elephant](elephant.png){width=4in, fig-alt="an alternative text string"}`

I have seen examples that place image-producing code in an R block, e.g.:
```
```{r}
#| width=4in
#| fig-alt='an alternative text string'
ggplot(something)
```
```
How would you load an external image file into a R block?

TIA

**Comment by mcanouil** [comment]
What do you call "extensions"?
What do you call "external"?

Why do you want to use an R code cell to include an image? There is no need for computations. Plain markdown is enough.

**Comment by ABSOD** [comment]
My understanding is that in this command:

`![Elephant](elephant.png){width=4in}
`
the  `{width=4in}`  statement is the extension.

I need to set multiple parameters for the images for accessibility reasons.  I need to control the size, and the alignment AND set a specific alt-text string for multiple images in the document.

How do I do this?
TIA

**Reply by ABSOD** [reply]
One more thing, I am generating multiple PDF documents in a loop.  Each document will need to include different images based on the values passed to the .qmd file through the kintr parameters.  I'm thinking I need to pass the image information in an R block.
Cheers

**Reply by mcanouil** [reply]
> the  `{width=4in}`  statement is the extension.

That is not an "extension". That is an attribute.
An "extension" for an image is `.png` from your example.
Could you update the title of your post which is very misleading? Thank you.

You can add any attributes as described in the documentation, see <https://quarto.org/docs/authoring/markdown-basics.html#links-images> and <https://quarto.org/docs/authoring/figures.html#alt-text>.
Note that the names of the attributes are the ones from HTML mostly and that they are space separated (no comma).

Regarding the loop, either output markdown using `cat()` and `output: asis` or use `knitr::include_graphics()`.
